src/special/mocha.js

```diff
@@ -1,41 +1,82 @@
 import fs from 'fs';
 import path from 'path';
-import lodash from 'lodash';
 import requirePackageName from 'require-package-name';
 import { getScripts } from '../utils';
 
-function getOpts(script) {
-  const argvs = script.split(' ').filter(argv => argv);
+const knownReporters = [
+  'dot', 'doc', 'tap', 'json', 'html', 'list',
+  'min', 'spec', 'nyan', 'xunit', 'markdown', 'progress',
+  'landing', 'json-stream',
+];
+
+function getOptsConfig(root, config) {
+  const argvs = config.split(/\s+/);
   const optsIndex = argvs.indexOf('--opts');
-  return optsIndex !== -1 ? argvs[optsIndex + 1] : null;
+
+  if (optsIndex === -1) {
+    return null;
+  }
+
+  const optsPath = argvs[optsIndex + 1];
+
+  if (!optsPath) {
+    return null;
+  }
+
+  return fs.readFileSync(path.resolve(root, '..', optsPath), 'utf-8');
```

Comment:
What if the file does not exist? The exception will break whole parser.

Reply:
what would be the correct thing to do? if they pass `--opts nonexistent`, we should throw or error in some way so they know they've tried to pass a non-existent config.